core/transport/src/main/proto/grpc/block_master.proto

```diff
@@ -223,14 +226,22 @@ message CommitBlockInUfsPRequest {
 message CommitBlockInUfsPOptions {}
 message CommitBlockInUfsPResponse {}
 
-message GetWorkerIdPOptions {}
+message GetWorkerIdPOptions {
+  optional bool extendedRegisterInfo = 1; // Identify whether there is preRegister information
+}
 message GetWorkerIdPRequest {
   /** the worker network address */
   optional grpc.WorkerNetAddress workerNetAddress = 1;
+  /** for preRegister information **/
   optional GetWorkerIdPOptions options = 2;
+  optional string clusterId = 3;
+  optional bool hasBlockInTier = 4;
 }
 message GetWorkerIdPResponse {
   optional int64 workerId = 1;
+  optional grpc.WorkerNetAddress workerNetAddress = 2;
+  optional WorkerPreRegisterInfo workerPreRegisterInfo = 3;
+  optional GetWorkerIdPOptions options = 4;
```

Comment:
Do you need the option in the response? The client already has the option when sending the request.